.github/workflows/ci.yml

```diff
@@ -2,170 +2,77 @@ name: CI
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@master
+      - uses: taiki-e/install-action@nextest
+      - name: nextest archive
+        run: cargo nextest archive --workspace --all-features --cargo-profile ci --archive-file 'nextest-archive-${{ matrix.platform.os }}.tar.zst'
+      - uses: actions/upload-artifact@v4
         with:
-          components: rustfmt
-          toolchain: nightly-2024-11-23
-      - uses: Swatinem/rust-cache@v2
-      - run: scripts/rust_fmt.sh --check
+          name: nextest-archive-${{ matrix.platform.os }}
+          path: nextest-archive-${{ matrix.platform.os }}.tar.zst
 
-  # Checks all .cairo files in the repo are formatted correctly.
-  cairofmt:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@stable
-      - uses: Swatinem/rust-cache@v2
-      - run: scripts/cairo_fmt.sh --check
-
-  # Checks all cairo corelib tests run correctly.
-  cairotest:
-    runs-on: ubuntu-latest
+  test:
+    name: test ${{ matrix.platform.name }} ${{ matrix.partition }}/8
+    runs-on: ${{ matrix.platform.os }}
+    needs:
+      - build-test
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - name: linux x86-64
+            os: ubuntu-latest
+        partition: [ 1, 2, 3, 4, 5, 6, 7, 8 ]
     steps:
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@stable
-      - uses: Swatinem/rust-cache@v2
-      - run: scripts/cairo_test.sh
-      - run: scripts/starknet_test.sh
-
-  # Check for unnecessary dependencies.
-  udeps:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v4
-      - name: Machete
-        uses: bnjbvr/cargo-machete@main
-
-  clippy:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@master
+      - uses: taiki-e/install-action@nextest
+      - uses: actions/download-artifact@v4
         with:
-          components: clippy
-          toolchain: nightly-2024-11-23
-      - uses: Swatinem/rust-cache@v2
-      - run: >
-          scripts/clippy.sh
+          name: nextest-archive-${{ matrix.platform.os }}
+      - name: nextest partition ${{ matrix.partition }}/8
+        run: cargo nextest run --partition 'count:${{ matrix.partition }}/8' --archive-file 'nextest-archive-${{ matrix.platform.os }}.tar.zst'
 
-  typos:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: crate-ci/typos@v1.27.0
-
-  docs:
+  check:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@master
         with:
           toolchain: nightly-2024-11-23
+          components: rustfmt
+      - uses: dtolnay/rust-toolchain@stable
       - uses: Swatinem/rust-cache@v2
-      - run: >
-          scripts/docs.sh
+      - run: cargo +nightly-2024-11-23 fmt --check
```

Comment:
why do we need the nightly here

Reply:
we're using nightly rustfmt settings, they live in separate section in rustfmt.toml